src/cli/restore/exchange.go

```diff
@@ -215,23 +216,22 @@ func restoreExchangeCmd(cmd *cobra.Command, args []string) error {
 
 	defer utils.CloseRepo(ctx, r)
 
+	dest := control.DefaultRestoreDestination(common.SimpleDateTime)
+
 	sel := utils.IncludeExchangeRestoreDataSelectors(opts)
 	utils.FilterExchangeRestoreInfoSelectors(sel, opts)
 
-	// if no selector flags were specified, get all data in the service.
-	if len(sel.Scopes()) == 0 {
-		sel.Include(sel.AllData())
-	}
-
-	restoreDest := control.DefaultRestoreDestination(common.SimpleDateTime)
-
-	ro, err := r.NewRestore(ctx, backupID, sel.Selector, restoreDest)
+	ro, err := r.NewRestore(ctx, backupID, sel.Selector, dest)
 	if err != nil {
 		return Only(ctx, errors.Wrap(err, "Failed to initialize Exchange restore"))
 	}
 
 	ds, err := ro.Run(ctx)
 	if err != nil {
+		if errors.Is(err, kopia.ErrNotFound) {
```

Comment:
this makes me pause a bit and feels like a code smell. I know that it means we're missing a backup or backup details, but it really feels like we're pulling in info from too low a layer just to say that. It feels like this message should be generated at the layer that actually handles the error and passed back up rather than pulling in the bottom layer just to check (and having the assumption that this layer knows what the bottom layer returns error-wise for this)

Reply:
> should be generated at the layer that actually handles the error and passed back up

I'm heavily against this pattern.  It's effectively the same as saying, "the lowest level error should decide the returned http code".  Seems like nice qol, but ultimately ends up coupling layers together by binding concerns across interfaces.  

What I think you mean to point out by noting a smell is that we're slicing the source error details to present a less detailed, but more easily observed, error message, and that this behavior reduces insight and debugging capacity.  That I absolutely agree with.  The ideal solution is to diverge end-user presentation from granular detail. That way we can keep presentation more coarse ("that backup id is unusable"), and not lose debugging insights ("the backup id doesn't have details data").  But ownership of that divergence should be at the cli/api layer, not at the error production.

Reply:
>ultimately ends up coupling layers together by binding concerns across interfaces.

I'm a bit confused by this statement. Isn't the current code doing the same though? It's effectively saying that stuff in the kopia package can't change their errors or they will break the CLI

Given that the CLI is mostly a wrapper around the SDK layer, it seems like maybe the SDK should be dealing with this (maybe that will be rolled into error management/logging?)

Reply:
> It's effectively saying that stuff in the kopia package can't change their errors or they will break the CLI

Not quite the same.  It's saying that if lower level packages change their errors, then the CLI might produce a less accurate message, being unable to specifically correlate that error to a sentinel.  Is it similar?  Yes.  But it's a lot more resilient to actual breaking, and keeps concerns appropriately separated that way.  

> it seems like maybe the SDK should be dealing with this

If by "dealing with" you mean "interpreting errors", I don't think that's the best choice.  Outside of, perhaps, organizing some is/as checks to be more easily consumable, I think we're meeting the needs of the sdk well enough.  Further interpretation/transformation feels like middleware for its own sake.

Reply:
mmmm alright. At this point it feels a bit like we're bikeshedding on encapsulation/information hiding at any rate

Reply:
Thoughts on just adding a call to `getBackupAndDetailsFromID` before the operation is Run and if that fails - returning the error below?

Reply:
Doesn't change much about the error handling, ultimately.  Plus it duplicates validation that we're already expecting `Run()` to cover.  I'd rather the runner maintain its own validation requirements as a centralized precheck than spread those requirements out to callers piecemeal.